### ISSUE TITLE
Automatically Pulls Agrro as Party Leader On Patch 1.1

### ISSUE THREAD

**Issue by androbourne** [tiltedphoques/TiltedEvolution#307]
No mods are required to reproduce this issue. Can be reproduced with no mods.

## Description

Even after patch 1.1 it appears mobs will still automatically aggro the party leaders. Regardless of you as party leader performing any actions such as damage by simply being close enough to the NPC.

The situation was a player was agrroed by a mob and attacking it. The second I as the party leader got within about 10 feet of the enemy it would redirect and aggro onto me instead. You do not need to hit the enemy or be in line of sight. Simply just being close enough is enough to steal aggro from non-party leaders.

Related formIDs|
none.

## Steps to reproduce

How to reproduce this issue. Be as specific as possible and try to only include instructions you believe are necessary to reproduce said issue, i.e.

1. Create a party
2. Invite a player
3. Have non party leader pull mob agrro
4. As party leader walk near the mob (roughly around 10ft or so)
5. Party leader automatically pulls mob agrro

## Reproduction rate

Please check just _one_ of the following reproduction rates: Can reproduce 100% of the time.

- [x] Consistent: I can reproduce this bug by following the steps to reproduce 100% of the time (*every time*)
- [ ] Occasional: I can reproduce this bug by following the steps to reproduce around 50% of the time
- [ ] Rare: I can reproduce this bug by following the steps to reproduce around 10% or less of the time

## Expected result

Expect result is that the party leader shouldn't automatically pull aggro for just being near the mob. Mob should be aggroed onto the players hitting him.

## Evidence (optional)

https://youtu.be/xhgxAmtFB0w?t=376

**Comment by MatthaeusDombrowski** [comment]
This same issue is happening to me in V1.2.
I believe that this bug deserves a higher priority, since it makes the gameplay really frustrating(the other party members can attack free of hits, while the party leader receives all of the damage)

**Comment by androbourne** [comment]
> This same issue is happening to me in V1.2. I believe that this bug deserves a higher priority, since it makes the gameplay really frustrating(the other party members can attack free of hits, while the party leader receives all of the damage)

I agree. The same goes for issue #306. They are really game-breaking issues.

**Comment by VRKuro** [comment]
In 100% agreement here. Makes things way more difficult when everything is just targeting the host when the person you're playing with is the tank :(

**Comment by RobbeBryssinck** [comment]
I agree that these 2 issues are most prominent right now, but with the main dev team being on hiatus, I don't see it getting fixed soon.

**Comment by MatthaeusDombrowski** [comment]
This needs more testing...
Now I tested creating the party before entering an dungeon(the way it is intended, as defined in the play guide) and everything worked just fine!!!

I opened two clients, one being the party leader, and made the non-leader take aggro first. The enemies attacked him and not switched to attack the leader as they did before.

I could even pull aggro towards the leader by, playing with him, dealing certain amount of damage to the enemies.

If someone other than me confirm this, it will be really nice

**Comment by androbourne** [comment]
> This needs more testing... Now I tested creating the party before entering an dungeon(the way it is intended, as defined in the play guide) and everything worked just fine!!!
> 
> I opened two clients, one being the party leader, and made the non-leader take aggro first. The enemies attacked him and not switched to attack the leader as they did before.
> 
> I could even pull aggro towards the leader by, playing with him, dealing certain amount of damage to the enemies.
> 
> If someone other than me confirm this, it will be really nice

I do not believe it needs more test. I can reproduce this 100% of the time without a doubt. Watch the video. We have played with different party leaders and the results are exactly the same. It is about party leader radius to the mob. If you are with in X distance of a mob, the party leader will automatically pull aggro. Even if the agrros was on a party member before hand.

I implore you to watch the video again. It is very easy to produce, even some of the modders have stated they were able to reproduce it and admit is a problem.

**Comment by MatthaeusDombrowski** [comment]
Ok ok, I believe you. Sorry if I didn't expressed myself well.

I have tested not because I read this issue, but because **this issue was happening to me.**
In my case, the problem occurs exactly like you described, and exactly like the video, but _it seems_ that only if the party is created after the players entered the dungeon/interior cell.

When creating the party before, and then entering, the enemies aggro as expected: they target the closest player(even if he is a non-leader and the leader is in their line of sight). Playing as the leader, I only was able to make the enemies change the aggro to me after I attacked then a couple of times, what makes total sense.

Basically, to me, everything worked just fine, but only if the party was created before the cell was loaded(I'm not sure how this works in exterior cells...)

But, obviously, it is possible that this will not work in every case/setup. So, I believe that more testing is appropriate, specially if done by several people.

**Comment by androbourne** [comment]
> Ok ok, I believe you. Sorry if I didn't expressed myself well.
> 
> I have tested not because I read this issue, but because **this issue was happening to me.** In my case, the problem occurs exactly like you described, and exactly like the video, but _it seems_ that only if the party is created after the players entered the dungeon/interior cell.
> 
> When creating the party before, and then entering, the enemies aggro as expected: they target the closest player(even if he is a non-leader and the leader is in their line of sight). Playing as the leader, I only was able to make the enemies change the aggro to me after I attacked then a couple of times, what makes total sense.
> 
> Basically, to me, everything worked just fine, but only if the party was created before the cell was loaded(I'm not sure how this works in exterior cells...)
> 
> But, obviously, it is possible that this will not work in every case/setup. So, I believe that more testing is appropriate, specially if done by several people.

I understand what you mean now. Got it. We created the party in the open world and teleported to that location, so we werent in that cell of the mob prior to the party being made and it still happened. Maybe it is based on the radius to which the party was created and the mobs. So that could indeed be the issue.

**Comment by kulcris** [comment]
I suspect the NPC is considering all non-owner players to be THE player instead of a follower, 

resulting in the NPCs targeting the location of the owner when the non owners are triggering the NPC AI to find THE target.

This obviously did not fix the issue, Can we revert this? 
https://github.com/tiltedphoques/TiltedEvolution/pull/277

**Comment by cliffclick** [comment]
Happens to my party of 3 fairly often.  I'm party lead and squishy mage; mobs consistently run by my 2 tanky buddies attempting to body block to no avail.  Complaints from heavy armor tanks, because they receive NO damage, hence no heavy armor perks.  Very frustrating.
